### PR TITLE
feat: update zaps for approved targets

### PR DESCRIPTION
**Pull request by frontier159**
# Description
What does this PR solve?

Fixes # (issue)

# Checklist
- [ ] Code follows the style guide
- [ ] I have performed a self-review of my own code
- [ ] New and existing tests pass locally
- [ ] This PR is targeting the correct branch

### CODE REVIEW

File: protocol/contracts/zaps/TempleZaps.sol [hunk 1]
```diff
@@ -232,7 +232,8 @@ contract TempleZaps is ZapBase {
     address token1 = IUniswapV2Pair(pair).token1();
 
     if (_fromAddress != token0 && _fromAddress != token1) {
-
+      require(approvedTargets[_fromAddress][_swapTarget], "TempleZaps: Unsupported token/target");
```
Comment (frontier159) [inline protocol/contracts/zaps/TempleZaps.sol:235]: @princetonbishop It's a little odd that the `approvedTargets` are on this TempleZaps (via the ZapBase inheritance) too, but currently unused.

What were you thinking here - that TempleZaps has it's own allow list, or that it shares the same list as GenericZap (in which case we'd change this line to use `zaps.approvedTargets()`?

File: protocol/contracts/zaps/GenericZap.sol [hunk 2]
```diff
@@ -192,7 +192,7 @@ contract GenericZap is ZapBase {
     address swapTarget,
```
Comment (frontier159) [inline protocol/contracts/zaps/GenericZap.sol:192]: At the top of this contract -- In order for this to work in testnet, would have to change:
```
  ICurveFactory private immutable curveFactory = ICurveFactory(0xB9fC157394Af804a3578134A6585C0dc9cc990d4);
  IBalancerVault private immutable balancerVault = IBalancerVault(0xBA12222222228d8Ba445958a75a0704d566BF2C8);
```
To be set in the constructor too...know what the goerli addresses are (if any)?

File: protocol/contracts/zaps/TempleZaps.sol (hunk 1 above)
Comment (frontier159) [reply protocol/contracts/zaps/TempleZaps.sol:235]: Also, I can't see any tests which check that the unsupported token/target fails...might be worth setting up solidity coverage in temple core?

File: protocol/contracts/zaps/GenericZap.sol (hunk 2 above)
Comment (princetonbishop) [reply protocol/contracts/zaps/GenericZap.sol:192]: `balancerVault` can be found here
https://goerli.etherscan.io/address/0xBA12222222228d8Ba445958a75a0704d566BF2C8

File: protocol/contracts/zaps/TempleZaps.sol (hunk 1 above)
Comment (princetonbishop) [reply protocol/contracts/zaps/TempleZaps.sol:235]: for consistency across zaps, making `templeZaps` use zaps approved targets is more ideal.

Comment (frontier159) [reply protocol/contracts/zaps/TempleZaps.sol:235]: Works for me -- should probably remove from ZapBase and put directly into GenericZaps then I think

File: protocol/contracts/zaps/GenericZap.sol (hunk 2 above)
Comment (frontier159) [reply protocol/contracts/zaps/GenericZap.sol:192]: Will 0x on goerli have issues routing to the right pools if they're self-created?

**Pull request closed without merging**